### ISSUE TITLE
Add C++ SDK

### ISSUE THREAD

**Issue by dicej** [fermyon/spin#657]
Per #78, we now have an example of writing a spin-http component in C++, but as of this writing it uses the `wit-bindgen`-generated C API, which does not lend itself to idiomatic, modern C++.  It would be nice to have a proper SDK for C++ which uses e.g. std::string, std::tuple, std::optional, etc. for interface types.

**Comment by Mossaka** [comment]
With #643 been discussed, would you like to prioritize the intermediate C SDK work first and then build C++ SDK out of it, or this is in parallel to that work?

**Comment by dicej** [comment]
I would prioritize #643 first since that's likely to be a more stable platform for building a C++ SDK than the `wit-bindgen`-generated bindings.
